Add Segments Explainer

ststimac opened the pull request:
Since the thinking for window segments for multi-screen devices evolved to be a `segments` property hanging off the visual viewport API, we're archiving the windows segments explainer and will point to this segments explainer that has been updated with the latest thinking, to be included here in the Visual Viewport repo.

This PR: 
Adds the Segments explainer to the repo
Fixes a typo in the draft spec

@dlibby-

ststimac committed:
Add segments-explainer

ststimac committed:
Merge pull request #4 from ststimac/ststimac-segments-explainer

Add segments explainer

ststimac committed:
Update image path

Update image path

Add SVGs

Update note in code for testing

Update SEGMENTS-EXPLAINER.md

Grammar edits, add name, remove Surface Neo reference

Update SEGMENTS-EXPLAINER.md

ststimac committed:
Update README with segments explainer

ststimac committed:
Fix spelling error

ststimac commented:
I'm having issues revalidating after linking my W3C account.

scottlow commented:
@dlibby @bokand, any feedback on this one before merging? It'd be great to have this one in so we can point the TAG to it for final review. Thanks!

dlibby- commented on segments-explainer/SEGMENTS-EXPLAINER.md:
````diff
@@ -0,0 +1,100 @@
+```js  
+const segments = window.visualViewport.segments;
+
+if( segments.length > 1 ) {
````
This should be updated to handle the null segments case instead of testing for length > 1

dlibby- commented on segments-explainer/SEGMENTS-EXPLAINER.md:
```diff
@@ -0,0 +1,100 @@
+if( segments.length > 1 ) {
+	// now we know the device is a foldable
+	// it's recommended to test whether segments[0].width === segments[1].width
+	// and we can update CSS classes in our layout as appropriate 
```
Can we link to the CSS specs (and/or give some examples) to tie these concepts together?

dlibby- commented on segments-explainer/SEGMENTS-EXPLAINER.md:
````diff
@@ -0,0 +1,100 @@
+}
+```
+
+The value returned from querying `visualViewport.segments` will be an array of DOMRects and based on the data returned for each "viewport segment", developers will be able to infer the number of hinges available as well as the hinge orientation. When `visualViewport.segments` is queried on a desktop, this will return `null` since there is only one viewport segment.
````
instead of 'on a desktop' I think we should say when there is only a single segment, and mention it is not exposed in this case because this information would be redundant with other fields on VisualViewport (and also could lead to future compat issues on these devices if authors started using visualViewport.segments[0] to target single-screen devices)

ststimac committed:
Merge branch 'WICG:gh-pages' into gh-pages

ststimac committed:
Address Daniel's changes

- Add CSS spec links
- Change JavaScript example to use null segments case
- Change "on desktop" wording to "one segment"

ststimac committed:
Fix markdown links

ststimac committed:
Update with previous changes

Updating the hardware section to remove reference to Neo and other minor verbiage.

bokand commented:
Sorry for the delay - I just got back from a leave.

I'm happy to merge this when @dlibby is happy with it - let me know.

dlibby- commented on segments-explainer/SEGMENTS-EXPLAINER.md:
````diff
@@ -0,0 +1,105 @@
+
+```js  
+window.onresize = function() {
+	const segments = window.visualViewport.segments;
````
Sorry for missing this before, but this needs the same treatment as the previous script block. Also, we probably want to have something like:

```js  
window.onresize = function() {
    const segments = window.visualViewport.segments;
    if (segments && segments.length > 1) {
        // Make changes two split content into the segments.
    } else {
        // Reset state to single viewport (normal responsive layout).
    }
}
```

dlibby- reviewed (changes requested):
LGTM with one remaining suggestion.

ststimac committed:
Update JS example

Updated per Daniel's code

ststimac committed:
Code update

ststimac commented:
Made @dlibby-'s update, so should be good to go now. Thanks!

dlibby- reviewed (approved):
LGTM, thanks!
